SharedSource/Development/UnitTesting/NUnit/NUnitExtensions.cs

```diff
@@ -0,0 +1,82 @@
+﻿// This file is part of the re-motion Core Framework (www.re-motion.org)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+//
+// The re-motion Core Framework is free software; you can redistribute it
+// and/or modify it under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation; either version 2.1 of the
+// License, or (at your option) any later version.
+//
+// re-motion is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+//
+
+using System;
+using NUnit.Framework.Constraints;
+using Remotion.Utilities;
+
+namespace Remotion.Development.UnitTesting.NUnit
+{
+  public static partial class NUnitExtensions
+  {
+    public static EqualConstraint ArgumentOutOfRangeExceptionMessageEqualTo (this ConstraintExpression constraintExpression, string message, string paramName, int actualValue)
+    {
+      AssertThatMessageContainsWhitespaces (message: message);
+      AssertThatParameterDoesNotContainWhitespaces (paramName: paramName);
+      return constraintExpression.With.Message.EqualTo (new ArgumentOutOfRangeException (message: message, paramName: paramName, actualValue: actualValue).Message);
+    }
+
+    public static ContainsConstraint ArgumentExceptionMessageContains (this ConstraintExpression constraintExpression, string message, string paramName)
```

Comment:
that one needs to be checked if it's needed.

Reply:
```
.With.ArgumentExceptionMessageContains (
                  "The 'PreparingRedirectToSubFunctionState' type only supports WxePermaUrlOptions with the UsePermaUrl-flag set to true.",
                  "parameters"));
```
it's only used once and the check could also be an `EqualTo`, because the whole message and the parameter are checked